### PR TITLE
Comms filter scaffolding

### PR DESCRIPTION
**Pull request by RobotCaleb**
This was one attempt. Looking into genning iteration through all managers.

### CODE REVIEW

File: src/component/gas_producer_component.h [hunk 1]
```diff
@@ -7,6 +7,7 @@
 
 #include "component_manager.h"
 #include "../enums/enums.h"
+#include "wire_filter.h"
```
Comment (chrisforbes) [inline src/component/gas_producer_component.h]: Squash into previous commit.

**Comment by RobotCaleb** [comment]
@chrisforbes should be ready to go. is based on #582, so may want that one first.

File: gen/templates/component_ui_genned.cc.gen [hunk 2]
```diff
@@ -0,0 +1,41 @@
+/* THIS FILE IS AUTOGENERATED BY gen/gen_comps.py; DO NOT HAND-MODIFY */
+
+#include <map>
+#include "../component/component_system_manager.h"
+
+extern component_system_manager component_system_man;
+
+std::map<const std::string, std::string> get_filter_map(c_entity entity) {
```
Comment (chrisforbes) [inline gen/templates/component_ui_genned.cc.gen]: vector

**Comment by chrisforbes** [comment]
Obsolete

**Pull request closed without merging**